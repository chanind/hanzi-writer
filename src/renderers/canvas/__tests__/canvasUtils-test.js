const { pathStringToCanvas } = require('../canvasUtils');

describe('canvasUtils', () => {
  let ctx;

  beforeEach(() => {
    ctx = document.createElement('canvas').getContext('2d');
  });

  describe('pathStringToCanvas', () => {
    const pathStrings = [
      'M 350 571 Q 380 593 449 614 Q 465 615 468 623 Q 471 633 458 643 Q 439 656 396 668 Q 381 674 370 672 Q 363 668 363 657 Q 364 621 200 527 Q 196 518 201 516 Q 213 516 290 546 Q 303 550 316 556 L 350 571 Z',
      'M 584 466 Q 666 485 734 497 Q 746 496 754 511 Q 755 524 729 533 Q 693 554 622 527 Q 598 520 575 511 L 537 499 Q 518 495 500 488 Q 442 472 386 457 L 337 446 Q 327 446 179 416 Q 148 409 173 392 Q 212 365 241 376 Q 287 389 339 404 L 387 416 Q 460 438 545 457 L 584 466 Z',
      'M 386 457 Q 387 493 398 517 Q 405 535 390 548 Q 371 564 350 571 C 323 583 303 583 316 556 Q 315 556 316 555 Q 338 519 337 478 Q 337 462 337 446 L 339 404 Q 340 343 339 289 L 338 241 Q 337 180 334 133 Q 333 115 323 109 Q 317 105 250 119 Q 238 122 239 114 Q 240 108 249 100 Q 309 42 328 6 Q 341 -10 357 3 Q 390 36 390 126 Q 387 169 387 265 L 387 306 Q 387 355 387 416 L 386 457 Z',
      'M 339 289 Q 254 261 161 229 Q 139 222 101 221 Q 86 220 85 207 Q 84 192 94 184 Q 119 166 157 147 Q 169 144 182 154 Q 239 199 338 241 L 387 265 Q 477 314 484 318 Q 499 327 498 337 Q 492 343 479 340 Q 434 324 387 306 L 339 289 Z',
      'M 635 195 Q 690 75 797 -14 Q 876 -62 898 -47 Q 920 -37 914 3 Q 905 34 899 152 Q 900 174 894 178 Q 890 179 884 160 Q 857 75 838 60 Q 823 56 785 88 Q 710 155 670 226 L 644 279 Q 599 381 584 466 L 575 511 Q 547 659 576 752 Q 586 779 543 805 Q 509 827 489 825 Q 470 824 479 795 Q 503 752 507 707 Q 517 601 537 499 L 545 457 Q 573 334 612 245 L 635 195 Z',
      'M 612 245 Q 558 197 452 138 Q 442 132 448 128 Q 455 124 468 126 Q 523 135 574 160 Q 608 175 635 195 L 670 226 Q 706 260 747 317 Q 762 336 778 354 Q 788 361 785 374 Q 781 386 753 410 Q 734 428 723 428 Q 708 427 707 411 Q 701 354 644 279 L 612 245 Z',
      'M 687 669 Q 718 648 754 623 Q 770 613 786 615 Q 798 618 801 632 Q 802 648 789 678 Q 780 697 746 708 Q 665 726 651 715 Q 647 711 651 697 Q 655 687 687 669 Z',
    ];

    pathStrings.forEach((pathString, index) => {
      it(`translates SVG pathstrings from MakeMeAHanzi into canvas commands case ${index}`, () => {
        const commandFunc = pathStringToCanvas(pathString);
        commandFunc(ctx);
        expect(ctx.__getEvents()).toMatchSnapshot();
      });
    });
  });
});
