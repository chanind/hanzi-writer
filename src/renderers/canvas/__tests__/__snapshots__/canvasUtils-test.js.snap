// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`canvasUtils pathStringToCanvas translates SVG pathstrings from MakeMeAHanzi into canvas commands case 0 1`] = `
Array [
  Object {
    "props": Object {},
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "beginPath",
  },
  Object {
    "props": Object {
      "x": 350,
      "y": 571,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "moveTo",
  },
  Object {
    "props": Object {
      "cpx": 380,
      "cpy": 593,
      "x": 449,
      "y": 614,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "cpx": 465,
      "cpy": 615,
      "x": 468,
      "y": 623,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "cpx": 471,
      "cpy": 633,
      "x": 458,
      "y": 643,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "cpx": 439,
      "cpy": 656,
      "x": 396,
      "y": 668,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "cpx": 381,
      "cpy": 674,
      "x": 370,
      "y": 672,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "cpx": 363,
      "cpy": 668,
      "x": 363,
      "y": 657,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "cpx": 364,
      "cpy": 621,
      "x": 200,
      "y": 527,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "cpx": 196,
      "cpy": 518,
      "x": 201,
      "y": 516,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "cpx": 213,
      "cpy": 516,
      "x": 290,
      "y": 546,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "cpx": 303,
      "cpy": 550,
      "x": 316,
      "y": 556,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "x": 350,
      "y": 571,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "lineTo",
  },
]
`;

exports[`canvasUtils pathStringToCanvas translates SVG pathstrings from MakeMeAHanzi into canvas commands case 1 1`] = `
Array [
  Object {
    "props": Object {},
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "beginPath",
  },
  Object {
    "props": Object {
      "x": 584,
      "y": 466,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "moveTo",
  },
  Object {
    "props": Object {
      "cpx": 666,
      "cpy": 485,
      "x": 734,
      "y": 497,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "cpx": 746,
      "cpy": 496,
      "x": 754,
      "y": 511,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "cpx": 755,
      "cpy": 524,
      "x": 729,
      "y": 533,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "cpx": 693,
      "cpy": 554,
      "x": 622,
      "y": 527,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "cpx": 598,
      "cpy": 520,
      "x": 575,
      "y": 511,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "x": 537,
      "y": 499,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "lineTo",
  },
  Object {
    "props": Object {
      "cpx": 518,
      "cpy": 495,
      "x": 500,
      "y": 488,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "cpx": 442,
      "cpy": 472,
      "x": 386,
      "y": 457,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "x": 337,
      "y": 446,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "lineTo",
  },
  Object {
    "props": Object {
      "cpx": 327,
      "cpy": 446,
      "x": 179,
      "y": 416,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "cpx": 148,
      "cpy": 409,
      "x": 173,
      "y": 392,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "cpx": 212,
      "cpy": 365,
      "x": 241,
      "y": 376,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "cpx": 287,
      "cpy": 389,
      "x": 339,
      "y": 404,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "x": 387,
      "y": 416,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "lineTo",
  },
  Object {
    "props": Object {
      "cpx": 460,
      "cpy": 438,
      "x": 545,
      "y": 457,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "x": 584,
      "y": 466,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "lineTo",
  },
]
`;

exports[`canvasUtils pathStringToCanvas translates SVG pathstrings from MakeMeAHanzi into canvas commands case 2 1`] = `
Array [
  Object {
    "props": Object {},
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "beginPath",
  },
  Object {
    "props": Object {
      "x": 386,
      "y": 457,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "moveTo",
  },
  Object {
    "props": Object {
      "cpx": 387,
      "cpy": 493,
      "x": 398,
      "y": 517,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "cpx": 405,
      "cpy": 535,
      "x": 390,
      "y": 548,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "cpx": 371,
      "cpy": 564,
      "x": 350,
      "y": 571,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "cpx1": 323,
      "cpx2": 303,
      "cpy1": 583,
      "cpy2": 583,
      "x": 316,
      "y": 556,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "bezierCurveTo",
  },
  Object {
    "props": Object {
      "cpx": 315,
      "cpy": 556,
      "x": 316,
      "y": 555,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "cpx": 338,
      "cpy": 519,
      "x": 337,
      "y": 478,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "cpx": 337,
      "cpy": 462,
      "x": 337,
      "y": 446,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "x": 339,
      "y": 404,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "lineTo",
  },
  Object {
    "props": Object {
      "cpx": 340,
      "cpy": 343,
      "x": 339,
      "y": 289,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "x": 338,
      "y": 241,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "lineTo",
  },
  Object {
    "props": Object {
      "cpx": 337,
      "cpy": 180,
      "x": 334,
      "y": 133,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "cpx": 333,
      "cpy": 115,
      "x": 323,
      "y": 109,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "cpx": 317,
      "cpy": 105,
      "x": 250,
      "y": 119,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "cpx": 238,
      "cpy": 122,
      "x": 239,
      "y": 114,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "cpx": 240,
      "cpy": 108,
      "x": 249,
      "y": 100,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "cpx": 309,
      "cpy": 42,
      "x": 328,
      "y": 6,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "cpx": 341,
      "cpy": -10,
      "x": 357,
      "y": 3,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "cpx": 390,
      "cpy": 36,
      "x": 390,
      "y": 126,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "cpx": 387,
      "cpy": 169,
      "x": 387,
      "y": 265,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "x": 387,
      "y": 306,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "lineTo",
  },
  Object {
    "props": Object {
      "cpx": 387,
      "cpy": 355,
      "x": 387,
      "y": 416,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "x": 386,
      "y": 457,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "lineTo",
  },
]
`;

exports[`canvasUtils pathStringToCanvas translates SVG pathstrings from MakeMeAHanzi into canvas commands case 3 1`] = `
Array [
  Object {
    "props": Object {},
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "beginPath",
  },
  Object {
    "props": Object {
      "x": 339,
      "y": 289,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "moveTo",
  },
  Object {
    "props": Object {
      "cpx": 254,
      "cpy": 261,
      "x": 161,
      "y": 229,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "cpx": 139,
      "cpy": 222,
      "x": 101,
      "y": 221,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "cpx": 86,
      "cpy": 220,
      "x": 85,
      "y": 207,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "cpx": 84,
      "cpy": 192,
      "x": 94,
      "y": 184,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "cpx": 119,
      "cpy": 166,
      "x": 157,
      "y": 147,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "cpx": 169,
      "cpy": 144,
      "x": 182,
      "y": 154,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "cpx": 239,
      "cpy": 199,
      "x": 338,
      "y": 241,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "x": 387,
      "y": 265,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "lineTo",
  },
  Object {
    "props": Object {
      "cpx": 477,
      "cpy": 314,
      "x": 484,
      "y": 318,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "cpx": 499,
      "cpy": 327,
      "x": 498,
      "y": 337,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "cpx": 492,
      "cpy": 343,
      "x": 479,
      "y": 340,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "cpx": 434,
      "cpy": 324,
      "x": 387,
      "y": 306,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "x": 339,
      "y": 289,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "lineTo",
  },
]
`;

exports[`canvasUtils pathStringToCanvas translates SVG pathstrings from MakeMeAHanzi into canvas commands case 4 1`] = `
Array [
  Object {
    "props": Object {},
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "beginPath",
  },
  Object {
    "props": Object {
      "x": 635,
      "y": 195,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "moveTo",
  },
  Object {
    "props": Object {
      "cpx": 690,
      "cpy": 75,
      "x": 797,
      "y": -14,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "cpx": 876,
      "cpy": -62,
      "x": 898,
      "y": -47,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "cpx": 920,
      "cpy": -37,
      "x": 914,
      "y": 3,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "cpx": 905,
      "cpy": 34,
      "x": 899,
      "y": 152,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "cpx": 900,
      "cpy": 174,
      "x": 894,
      "y": 178,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "cpx": 890,
      "cpy": 179,
      "x": 884,
      "y": 160,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "cpx": 857,
      "cpy": 75,
      "x": 838,
      "y": 60,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "cpx": 823,
      "cpy": 56,
      "x": 785,
      "y": 88,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "cpx": 710,
      "cpy": 155,
      "x": 670,
      "y": 226,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "x": 644,
      "y": 279,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "lineTo",
  },
  Object {
    "props": Object {
      "cpx": 599,
      "cpy": 381,
      "x": 584,
      "y": 466,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "x": 575,
      "y": 511,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "lineTo",
  },
  Object {
    "props": Object {
      "cpx": 547,
      "cpy": 659,
      "x": 576,
      "y": 752,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "cpx": 586,
      "cpy": 779,
      "x": 543,
      "y": 805,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "cpx": 509,
      "cpy": 827,
      "x": 489,
      "y": 825,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "cpx": 470,
      "cpy": 824,
      "x": 479,
      "y": 795,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "cpx": 503,
      "cpy": 752,
      "x": 507,
      "y": 707,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "cpx": 517,
      "cpy": 601,
      "x": 537,
      "y": 499,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "x": 545,
      "y": 457,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "lineTo",
  },
  Object {
    "props": Object {
      "cpx": 573,
      "cpy": 334,
      "x": 612,
      "y": 245,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "x": 635,
      "y": 195,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "lineTo",
  },
]
`;

exports[`canvasUtils pathStringToCanvas translates SVG pathstrings from MakeMeAHanzi into canvas commands case 5 1`] = `
Array [
  Object {
    "props": Object {},
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "beginPath",
  },
  Object {
    "props": Object {
      "x": 612,
      "y": 245,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "moveTo",
  },
  Object {
    "props": Object {
      "cpx": 558,
      "cpy": 197,
      "x": 452,
      "y": 138,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "cpx": 442,
      "cpy": 132,
      "x": 448,
      "y": 128,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "cpx": 455,
      "cpy": 124,
      "x": 468,
      "y": 126,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "cpx": 523,
      "cpy": 135,
      "x": 574,
      "y": 160,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "cpx": 608,
      "cpy": 175,
      "x": 635,
      "y": 195,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "x": 670,
      "y": 226,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "lineTo",
  },
  Object {
    "props": Object {
      "cpx": 706,
      "cpy": 260,
      "x": 747,
      "y": 317,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "cpx": 762,
      "cpy": 336,
      "x": 778,
      "y": 354,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "cpx": 788,
      "cpy": 361,
      "x": 785,
      "y": 374,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "cpx": 781,
      "cpy": 386,
      "x": 753,
      "y": 410,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "cpx": 734,
      "cpy": 428,
      "x": 723,
      "y": 428,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "cpx": 708,
      "cpy": 427,
      "x": 707,
      "y": 411,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "cpx": 701,
      "cpy": 354,
      "x": 644,
      "y": 279,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "x": 612,
      "y": 245,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "lineTo",
  },
]
`;

exports[`canvasUtils pathStringToCanvas translates SVG pathstrings from MakeMeAHanzi into canvas commands case 6 1`] = `
Array [
  Object {
    "props": Object {},
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "beginPath",
  },
  Object {
    "props": Object {
      "x": 687,
      "y": 669,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "moveTo",
  },
  Object {
    "props": Object {
      "cpx": 718,
      "cpy": 648,
      "x": 754,
      "y": 623,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "cpx": 770,
      "cpy": 613,
      "x": 786,
      "y": 615,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "cpx": 798,
      "cpy": 618,
      "x": 801,
      "y": 632,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "cpx": 802,
      "cpy": 648,
      "x": 789,
      "y": 678,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "cpx": 780,
      "cpy": 697,
      "x": 746,
      "y": 708,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "cpx": 665,
      "cpy": 726,
      "x": 651,
      "y": 715,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "cpx": 647,
      "cpy": 711,
      "x": 651,
      "y": 697,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
  Object {
    "props": Object {
      "cpx": 655,
      "cpy": 687,
      "x": 687,
      "y": 669,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "quadraticCurveTo",
  },
]
`;
